utils/trivialize-rules/trivialize-targets.js

```diff
@@ -0,0 +1,256 @@
+"use strict";
+
+let util = require("util");
+let path = require("path");
+let xml2js = require("xml2js");
+
+let fs = require("graceful-fs");
+let readdir = util.promisify(fs.readdir);
+let readFile = util.promisify(fs.readFile);
+let parseString = util.promisify(xml2js.parseString);
+
+let chalk = require("chalk");
+let validUrl = require("valid-url");
+let escapeStringRegexp = require("escape-string-regexp");
+let { explodeRegExp, UnsupportedRegExp } = require("./explode-regexp");
+
+let rulesDir = "src/chrome/content/rules";
+
+(async () => {
+  let filenames = (await readdir(rulesDir)).filter(fn => fn.endsWith(".xml"));
+  let filePromises = filenames.map(async filename => {
+    function createTag(tagName, colour, print) {
+      return (strings, ...values) => {
+        let result = `[${tagName}] ${chalk.bold(filename)}: ${strings[0]}`;
+        for (let i = 1; i < strings.length; i++) {
+          let value = values[i - 1];
+          if (value instanceof Set) {
+            value = Array.from(value);
```

Comment:
`value = [...value]` has the same result.

Reply:
I would like to keep this part of code somewhat sync-ed with `trivialize-rules.js`. If they are functionally equivalent, I am not going to change it. but thanks anyway :smile:

Reply:
@cschanaj Can you move this bit into a library?

Reply:
this function works only within the closure. if we move it into a library, I am afraid we need to change the function prototype. I would like to avoid making the unrelated changes given the logics of the script in this PR is already very complicated.

Reply:
@cschanaj Thanks for explaining.